.circleci/config.yml

```diff
@@ -122,7 +121,7 @@ jobs:
           key: cargo-v1-{{ arch }}-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
```

Comment:
I did not go through the whole config,yml - but I thought that the `cache-key` needs to be updated for a new cache to be started (and, potentially/hopefully, for the old one to be pruned).

Reply:
I believe CircleCI caches are immutable: https://circleci.com/docs/caching/#writing-to-the-cache-in-workflows
"Clearing" a cache requires using a new entry; the old one will be removed after the retention period.
This config is basically updating the cache by using a new cache key, every time the dependencies change.
There is room for pruning what is saved in CircleCI (especially for Rust dependencies, but we'd have to build that)